### PR TITLE
V1

### PR DESCRIPTION
**Pull request by sir-dunxalot**
Proposed API for version 1.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -319,78 +265,65 @@ module.exports = {
...
+    if (jsOptions.concatScripts && jsOptions.preserveOriginal) {
+      removeFromTree(scriptInputPaths);
+    }
 
-    /* Remove the unnecessary files */
-    if (!this.preserveOriginals) {
-      workingTree = fileRemover(workingTree, {
-        files: scriptInputFiles.concat(styleInputFiles)
-      });
+    if (cssOptions.concatScripts && cssOptions.preserveOriginal) {
```
Comment (jschilli) [inline index.js]: should that be `!cssOptions.preserveOriginal`

File: index.js [hunk 2]
```diff
@@ -319,78 +265,65 @@ module.exports = {
...
-      outputFile: outputPath + '.css',
-      wrapInFunction: false
-    });
+    workingTree = mergeTrees(trees.filter(function(e) { return e; }));
 
-    /* Combine all the files into the project's tree */
+    /* Remove the unnecessary script and style files */
 
-    var workingTree = mergeTrees([tree, concatenatedScripts, concatenatedStyles]);
+    if (jsOptions.concatScripts && jsOptions.preserveOriginal) {
```
Comment (jschilli) [inline index.js]: same here - not sure if the logic for preserveOriginal is correct - still grokking the context

**Comment by jschilli** [comment]
@sir-dunxalot this looks good - outside of my comment on the preserveOriginal logic.

the only wishlist item I had in the original request (but was obfuscated) was output filename per type - one name for css, one for js - I totally recognize that it's an edge case (porting legacy apps to ember-cli etc)

so may be cleaner to blow that off.

**Comment by sir-dunxalot** [comment]
Oh yes - I forgot that, sorry. I did an early attempt but the initial code structure didn't allow for it. This refactoring definitely does though - I'd be down to add it as an option.

I discovered one unrelated, outstanding issue with file concatenation at Ember CLI's `/tests` route so I'm still working on fixing that.

I'll integrate both these fixes plus your PR comment fixes shortly so we can release.

**Comment by sir-dunxalot** [comment]
Also, I'm not really sure how to write tests for this given it's all about changing environment and Broccoli settings.

**Comment by jschilli** [comment]
I've struggled with writing tests for Brocfile level options as well

I ended up going with straightup mocha testing for `ember-cli-bless` and borrowed helpers from ember-cli itself: 

https://github.com/jschilli/ember-cli-bless/blob/master/tests/acceptance/ember-cli-bless-test.js#L56

**Pull request merged**